core/src/main/scala/cats/syntax/applicativeError.scala

```diff
@@ -3,12 +3,18 @@ package syntax
 
 import cats.data.{Xor, XorT}
 
-trait ApplicativeErrorSyntax {
+trait ApplicativeErrorSyntax1 {
+  implicit def applicativeErrorSyntax[F[_], E, A](fa: F[A])(implicit F: ApplicativeError[F, E]): ApplicativeErrorOps[F, E, A] =
+    new ApplicativeErrorOps[F, E, A](fa)
+}
+
+trait ApplicativeErrorSyntax extends ApplicativeErrorSyntax1 {
   implicit def applicativeErrorIdSyntax[E](e: E): ApplicativeErrorIdOps[E] =
     new ApplicativeErrorIdOps(e)
 
-  implicit def applicativeErrorSyntax[F[_, _], E, A](fa: F[E, A])(implicit F: ApplicativeError[F[E, ?], E]): ApplicativeErrorOps[F[E, ?], E, A] =
+  implicit def applicativeErrorSyntax2[F[_, _], E, A](fa: F[E, A])(implicit F: ApplicativeError[F[E, ?], E]): ApplicativeErrorOps[F[E, ?], E, A] =
```

Comment:
@julienrf, being consistent is a good point although I would imagine you want to cross build this code, meaning, I am not sure you can to simply remove the `Unapply` usage in the code base.  
@ceedubs mentioned in the chatroom that we don't want to over invest in `Unapply` post the SI-2712 fix. That being said, I am totally willing to turn this into `Unapply`, if that's the consensus. 
Maintainers any thoughts?